### PR TITLE
only record per state change processing times for streaming pipelines

### PR DESCRIPTION
**Pull request by clmccart**
only record per state change processing times for streaming pipelines and not for batch pipelines

### CODE REVIEW

**Comment by clmccart** [comment]
@tudorm

**Comment by clmccart** [comment]
from @tudorm: https://screenshot.googleplex.com/6AfiyxSbgB3BDHX

**Comment by clmccart** [comment]
@jrmccluskey mind taking a look?

**Review by jrmccluskey — APPROVED** [review]
LGTM

**Comment by jrmccluskey** [comment]
Might also try to loop in someone with a Java background to give a quick check before merging

**Comment by clmccart** [comment]
> Might also try to loop in someone with a Java background to give a quick check before merging

@m-trieu mind taking a look?

**Comment by m-trieu** [comment]
Ack taking a look

File: runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/DataflowExecutionContextTest.java [hunk 1]
```diff
@@ -184,4 +195,39 @@ public void testDataflowExecutionStateTrackerRecordsCompletedProcessingTimes()
     Assert.assertEquals(
         expectedMetadata.userStepName(), tracker.getActiveMessageMetadata().get().userStepName());
   }
+
+  @Test
+  public void testDataflowExecutionStateTrackerDoesNotRecordCompletedProcessingTimesForBatch()
```
Comment (m-trieu) [inline runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/DataflowExecutionContextTest.java:200]: do we need to add a test to show that it does record for streaming?

**Review by m-trieu — APPROVED** [review]
LGTM except for maybe adding that test for streaming

Good since we only branch or not based on Streaming.  We do this a lot in the code where we branch based on streaming or not, which eventually turns to if some option is set or not.  If we keep adding logic branches that is only streaming or batch specific and impls diverge it might be a good idea to start introducing interfaces/abstract classes to break up the different impls.  Not needed now but in the future.

File: runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/DataflowExecutionContextTest.java (hunk 1 above)
Comment (clmccart) [reply runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/DataflowExecutionContextTest.java:200]: this is implicitly covered in the pre-existing tests the were written for the original CL

**Pull request merged**